### PR TITLE
modify senario where dataset has no mask provided

### PR DESCRIPTION
**Pull request by CoinCheung**
Hi,

As reported in [this](https://github.com/facebookresearch/maskrcnn-benchmark/issues/809) issue, when there is no `segmentation` field in the json file (which I believe will be quite common in practice, since most people would like to use this codebase in the bbox detection task), there will be a error of `index out of range` in the `boxlist` indexing operations. I feel maybe this modification would allow to get rid of this error by not adding the filed of `mask` to the `bboxlist` object.

I am not sure whether this modification is in consistent with  the designing spirit of this repo. If it is not proper to implement this modification, please let me know and I will delete this pr so that we can discuss another place where I feel maybe a little modification will make it better :)

### CODE REVIEW

File: maskrcnn_benchmark/data/datasets/coco.py [hunk 1]
```diff
@@ -79,9 +79,11 @@ def __getitem__(self, idx):
         classes = torch.tensor(classes)
         target.add_field("labels", classes)
 
-        masks = [obj["segmentation"] for obj in anno]
-        masks = SegmentationMask(masks, img.size, mode='poly')
-        target.add_field("masks", masks)
+        masks = [obj["segmentation"] for obj in anno if any(obj["segmentation"])]
```
Comment (fmassa) [inline maskrcnn_benchmark/data/datasets/coco.py]: maybe you could just add something similar to what is done for keypoints just below?
```
if anno and 'segmentation' in anno[0]:
    # rest of the code here
```

Comment (CoinCheung) [reply maskrcnn_benchmark/data/datasets/coco.py]: Hi, 

Thanks for this suggestion, I have modified the code as the keypoint part. Please add more comments if further modification is required.
